src/ceph.in

```diff
@@ -533,24 +536,38 @@ def do_command(parsed_args, target, cmdargs, sigdict, inbuf, verbose):
     else:
         return -errno.EINVAL, '', 'invalid command'
 
+    next_header_print = 0
+    # Set extra options for polling commands only:
+    if is_polling(valid_dict):
+        valid_dict['width'] = Termsize().cols
     while True:
-        ret, outbuf, outs = json_command(cluster_handle, target=target, argdict=valid_dict,
-                inbuf=inbuf)
-        if 'poll' not in valid_dict or not valid_dict['poll']:
-            # Don't print here if it's not a polling command
-            break
-        if ret:
-            ret = abs(ret)
-            print('Error: {0} {1}'.format(ret, errno.errorcode.get(ret, 'Unknown')),
-                    file=sys.stderr)
-            break
-        if outbuf:
-            print(outbuf.decode('utf-8'))
-        if outs:
-            print(outs, file=sys.stderr)
-        if parsed_args.period <= 0:
-            break
-        sleep(parsed_args.period)
+        try:
+            if next_header_print == 0 and is_polling(valid_dict):
+                valid_dict['print_header'] = True
+                next_header_print = Termsize().rows - 3
+            next_header_print -= 1
+            ret, outbuf, outs = json_command(cluster_handle, target=target,
+                argdict=valid_dict, inbuf=inbuf)
+            if is_polling(valid_dict):
+                valid_dict['print_header'] = False
+            if 'poll' not in valid_dict or not valid_dict['poll']:
```

Comment:
should be
```py
else:
  break
```

Reply:
@tchaikov Right, I thought about it, but the logic is different. The first `if` is for `print_header`, the second `if` is for deciding whether we print the output now or not. Since they're unrelated, I left them as two `if` statements to make the reading less confusing. I'm happy to change it though if you think it's better.